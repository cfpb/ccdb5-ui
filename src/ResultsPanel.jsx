import { connect } from 'react-redux'
import ListPanel from './ListPanel'
import MapPanel from './MapPanel'
import React from 'react'

export class ResultsPanel extends React.Component {
  render() {
    let currentPanel

    switch ( this.props.tab ) {
      case 'List':
        currentPanel = <ListPanel/>
        break;
      case 'Map':
      default:
        currentPanel = <MapPanel/>
        break;
    }

    return (
      <div className="content_main">
        { currentPanel }
      </div>
    )
  }
}

const mapStateToProps = state => ( {
  tab: state.query.tab
} )

export default connect( mapStateToProps )( ResultsPanel )
