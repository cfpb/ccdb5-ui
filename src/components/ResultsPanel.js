import { MODE_LIST, MODE_MAP, MODE_TRENDS } from '../constants';
import { useSelector } from 'react-redux';
import { ListPanel } from './List/ListPanel/ListPanel';
import { MapPanel } from './Map/MapPanel';
import { PrintInfo } from './Print/PrintInfo';
import { PrintInfoFooter } from './Print/PrintInfoFooter';
import TrendsPanel from './Trends/TrendsPanel';
import { selectQueryTab } from '../reducers/query/selectors';

export const ResultsPanel = () => {
  const tab = useSelector(selectQueryTab);
  let currentPanel;

  switch (tab) {
    case MODE_MAP:
      currentPanel = <MapPanel />;
      break;
    case MODE_LIST:
      currentPanel = <ListPanel />;
      break;
    case MODE_TRENDS:
    default:
      currentPanel = <TrendsPanel />;
      break;
  }

  return (
    <div className={'content__main ' + tab.toLowerCase()}>
      <PrintInfo />
      {currentPanel}
      <PrintInfoFooter />
    </div>
  );
};
