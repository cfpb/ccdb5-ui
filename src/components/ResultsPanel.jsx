import { MODE_LIST, MODE_MAP, MODE_TRENDS } from '../constants'
import { printModeOff, printModeOn } from '../actions/view'
import { connect } from 'react-redux'
import ListPanel from './List/ListPanel'
import MapPanel from './Map/MapPanel'
import PrintInfo from './Print/PrintInfo'
import PrintInfoFooter from './Print/PrintInfoFooter'
import React from 'react'
import TrendsPanel from './Trends/TrendsPanel'

export class ResultsPanel extends React.Component {
  constructor( props ) {
    super( props )
    // Bindings
    this._updatePrintStyleOn = this._togglePrintStylesOn.bind( this );
    this._updatePrintStyleOff = this._togglePrintStylesOff.bind( this );
  }

  componentDidMount() {
    window.addEventListener( 'afterprint', this._updatePrintStyleOff );
    window.addEventListener( 'beforeprint', this._updatePrintStyleOn );
  }

  componentWillUnmount() {
    window.removeEventListener( 'afterprint', this._updatePrintStyleOff );
    window.removeEventListener( 'beforeprint', this._updatePrintStyleOn );
  }

  _getTabClass() {
    const classes = [ 'content_main', this.props.tab.toLowerCase() ]
    return classes.join( ' ' )
  }
  /* eslint complexity: ["error", 6] */
  render() {
    let currentPanel

    switch ( this.props.tab ) {
      case MODE_MAP:
        currentPanel = <MapPanel/>
        break
      case MODE_LIST:
        currentPanel = <ListPanel/>
        break
      case MODE_TRENDS:
      default:
        currentPanel = <TrendsPanel/>
        break;
    }

    return (
      <div className={ this._getTabClass() }>
        { this.props.printMode && <PrintInfo/> }
        { currentPanel }
        { this.props.printMode && <PrintInfoFooter/> }
      </div>
    )
  }

  _togglePrintStylesOn() {
    const compProps = this.props;
    compProps.togglePrintModeOn();
  }
  _togglePrintStylesOff() {
    const compProps = this.props;
    compProps.togglePrintModeOff();
  }
}

const mapStateToProps = state => ( {
  printMode: state.view.printMode,
  tab: state.query.tab
} )

export const mapDispatchToProps = dispatch => ( {
  togglePrintModeOn: () => {
    dispatch( printModeOn() )
  },
  togglePrintModeOff: () => {
    dispatch( printModeOff() )
  }

} )

export default connect( mapStateToProps, mapDispatchToProps )( ResultsPanel )
