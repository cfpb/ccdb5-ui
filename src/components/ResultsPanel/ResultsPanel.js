import { MODE_LIST, MODE_MAP, MODE_TRENDS } from '../../constants';
import { useSelector } from 'react-redux';
import { ListPanel } from '../List/ListPanel/ListPanel';
import { MapPanel } from '../Map/MapPanel';
import { PrintInfo } from '../Print/PrintInfo';
import { PrintInfoFooter } from '../Print/PrintInfoFooter';
import { selectViewTab } from '../../reducers/view/selectors';
import { TrendsPanel } from '../Trends/TrendsPanel/TrendsPanel';

export const ResultsPanel = () => {
  const tab = useSelector(selectViewTab);
  let currentPanel;

  switch (tab) {
    case MODE_MAP:
      currentPanel = <MapPanel />;
      break;
    case MODE_LIST:
      currentPanel = <ListPanel />;
      break;
    case MODE_TRENDS:
    default:
      currentPanel = <TrendsPanel />;
      break;
  }

  return (
    <div className={'content__main ' + tab.toLowerCase()}>
      <PrintInfo />
      {currentPanel}
      <PrintInfoFooter />
    </div>
  );
};
