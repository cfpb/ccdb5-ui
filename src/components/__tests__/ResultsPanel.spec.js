import { mapDispatchToProps, ResultsPanel } from '../ResultsPanel';
import React from 'react';
import { shallow } from 'enzyme';

describe('component:Results', () => {
  let target;
  const actionMock = jest.fn();

  describe('event listeners', () => {
    beforeEach(() => {
      window.addEventListener = jest.fn();
      window.removeEventListener = jest.fn();
    });

    it('unregisters the same listener on unmount', () => {
      const a = window.addEventListener;
      const b = window.removeEventListener;

      target = shallow(<ResultsPanel tab="foo" />);
      expect(a.mock.calls.length).toBe(2);
      expect(a.mock.calls[0][0]).toBe('afterprint');
      expect(a.mock.calls[1][0]).toBe('beforeprint');

      target.unmount();
      expect(b.mock.calls.length).toBe(2);
      expect(b.mock.calls[0][0]).toBe('afterprint');
      expect(b.mock.calls[1][0]).toBe('beforeprint');

      expect(a.mock.calls[0][1]).toBe(b.mock.calls[0][1]);
    });
  });

  describe('print mode', () => {
    afterEach(() => {
      jest.clearAllMocks();
    });
    it('toggles print mode on', () => {
      target = shallow(
        <ResultsPanel togglePrintModeOn={actionMock} tab="Foobar" />
      );
      const instance = target.instance();
      instance._togglePrintStylesOn();
      expect(actionMock).toHaveBeenCalledTimes(1);
    });

    it('toggles print mode off', () => {
      target = shallow(
        <ResultsPanel togglePrintModeOff={actionMock} tab="Foobar" />
      );
      const instance = target.instance();
      instance._togglePrintStylesOff();
      expect(actionMock).toHaveBeenCalledTimes(1);
    });
  });

  describe('mapDispatchToProps', () => {
    afterEach(() => {
      jest.clearAllMocks();
    });
    it('provides a way to call togglePrintModeOn', () => {
      const dispatch = jest.fn();
      mapDispatchToProps(dispatch).togglePrintModeOn();
      expect(dispatch.mock.calls.length).toEqual(1);
    });
    it('provides a way to call togglePrintModeOff', () => {
      const dispatch = jest.fn();
      mapDispatchToProps(dispatch).togglePrintModeOff();
      expect(dispatch.mock.calls.length).toEqual(1);
    });
  });
});
