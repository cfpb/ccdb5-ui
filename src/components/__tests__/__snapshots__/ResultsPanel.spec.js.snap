// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component:Results renders Map print mode without crashing 1`] = `
<div
  className="content_main map"
>
  <section
    className="print-info"
  >
    <p>
      <span>
        Dates:
      </span>
       
      7/10/2017 - 7/10/2020
    </p>
    <div>
      <div>
        Showing 
        <span>
          1
        </span>
         out of
        <span>
           
          100
           
        </span>
         total complaints
         
      </div>
    </div>
  </section>
  <section
    className="map-panel"
  >
    <div>
      <summary
        className="action-bar"
        id="search-summary"
      >
        <div>
          <h2>
            Showing 
            1
             matches out of 
            100
             total complaints
          </h2>
        </div>
        <div>
          <h3
            className="h4 flex-all export-results"
          >
            <button
              className="a-btn a-btn__link export-btn"
              data-gtm_ignore="true"
              onClick={[Function]}
            >
              Export data
            </button>
            <button
              className="a-btn a-btn__link print-preview"
              onClick={[Function]}
            >
              <svg
                className="cf-icon-svg"
                viewBox="0 0 761.9 1200"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M203.7 889.9h355v50h-355zM203.7 788.4h355v50h-355z"
                />
                <path
                  d="M670.9 368.4L477.3 174.8H122.1c-16.5 0-30 13.5-30 30V500H70c-38.5 0-70 31.5-70 70v305.5c0 38.5 31.5 70 70 70h22.1v60.5c0 16.5 13.5 30 30 30h518.8c16.5 0 30-13.5 30-30v-60.5h21c38.5 0 70-31.5 70-70V569.9c0-38.5-31.5-70-70-70h-21V368.4zm-60 607.6H152.1V742.7h458.8V976zM425 234.8v146.4c0 21.1 17.3 38.4 38.4 38.4h147.5V500H152.1V234.8H425zm237.9 403.9c-16.6 0-30-13.4-30-30s13.4-30 30-30 30 13.4 30 30-13.4 30-30 30z"
                />
              </svg>
              Print
            </button>
          </h3>
        </div>
      </summary>
    </div>
    <div
      className="tabbed-navigation"
      data-tour="tabbed-navigation"
    >
      <section>
        <button
          className="trends tab"
          onClick={[Function]}
        >
          <svg
            className="cf-icon-svg"
            viewBox="0 0 15 19"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M13.55 15.256H1.45a.554.554 0 0 1-.553-.554V3.168a.554.554 0 1 1 1.108 0v10.98h11.544a.554.554 0 0 1 0 1.108zM3.121 13.02V6.888a.476.476 0 0 1 .475-.475h.786a.476.476 0 0 1 .475.475v6.132zm2.785 0V3.507a.476.476 0 0 1 .475-.475h.786a.476.476 0 0 1 .475.475v9.513zm2.785 0V6.888a.476.476 0 0 1 .475-.475h.786a.476.476 0 0 1 .475.475v6.132zm2.786 0v-2.753a.476.476 0 0 1 .475-.475h.785a.476.476 0 0 1 .475.475v2.753z"
            />
          </svg>
          Trends
        </button>
        <button
          className="list tab"
          onClick={[Function]}
        >
          <svg
            className="cf-icon-svg"
            viewBox="0 0 14 19"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M2.85 4.836a1.03 1.03 0 0 1-1.029 1.03h-.003a1.03 1.03 0 1 1 0-2.059h.003a1.03 1.03 0 0 1 1.03 1.03zm0 4.346a1.03 1.03 0 0 1-1.029 1.03h-.003a1.03 1.03 0 1 1 0-2.059h.003a1.03 1.03 0 0 1 1.03 1.03zm0 4.346a1.03 1.03 0 0 1-1.029 1.03h-.003a1.03 1.03 0 1 1 0-2.059h.003a1.03 1.03 0 0 1 1.03 1.03zm1.13-8.692a1.03 1.03 0 0 1 1.029-1.029h7.173a1.03 1.03 0 1 1 0 2.058H5.009a1.03 1.03 0 0 1-1.03-1.029zm9.231 4.347a1.03 1.03 0 0 1-1.029 1.029H5.009a1.03 1.03 0 1 1 0-2.059h7.173a1.03 1.03 0 0 1 1.03 1.03zm0 4.345a1.03 1.03 0 0 1-1.029 1.03H5.009a1.03 1.03 0 1 1 0-2.059h7.173a1.03 1.03 0 0 1 1.03 1.03z"
            />
          </svg>
          List
        </button>
        <button
          className="map tab active"
          onClick={[Function]}
        >
          <svg
            className="cf-icon-svg"
            viewBox="0 0 16 19"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15.084 15.2H.916a.264.264 0 0 1-.254-.42l2.36-4.492a.865.865 0 0 1 .696-.42h.827a9.51 9.51 0 0 0 .943 1.108H3.912l-1.637 3.116h11.45l-1.637-3.116h-1.34a9.481 9.481 0 0 0 .943-1.109h.591a.866.866 0 0 1 .696.421l2.36 4.492a.264.264 0 0 1-.254.42zM11.4 7.189c0 2.64-2.176 2.888-3.103 5.46a.182.182 0 0 1-.356 0c-.928-2.572-3.104-2.82-3.104-5.46a3.282 3.282 0 0 1 6.563 0zm-1.86-.005a1.425 1.425 0 1 0-1.425 1.425A1.425 1.425 0 0 0 9.54 7.184z"
            />
          </svg>
          Map
        </button>
      </section>
    </div>
    <div
      className="layout-row refine-bar"
    >
      <section
        className="filter-panel-toggle"
      >
        <div
          className="m-btn-group"
        >
          <p>
             
          </p>
          <button
            className="a-btn"
            onClick={[Function]}
          >
            Filter results
          </button>
        </div>
      </section>
      <span
        className="separator"
      />
      <section
        className="m-btn-group"
      >
        <p>
          Map shading
        </p>
        <button
          aria-label="Display map by complaints"
          className="a-btn "
          disabled={false}
          onClick={[Function]}
        >
          Complaints
        </button>
        <button
          aria-label="Display map by complaints per 1,000 people"
          className="a-btn a-btn__disabled"
          disabled={false}
          onClick={[Function]}
        >
          Complaints per 1,000 
          <span>
            population
          </span>
        </button>
      </section>
    </div>
    <div>
      <div
        className="cfpb-chart print"
        data-chart-type="tile_map"
        id="tile-chart-map"
      />
    </div>
    <div
      className="map-toolbar"
    >
      <section
        className="state-heading"
      >
        <span>
          United States of America
        </span>
        <span>
          
        </span>
        
      </section>
      
    </div>
    <div
      className="row-chart-section"
    >
      <h3>
        Product by highest complaint volume 7/10/2017 to 7/10/2020
      </h3>
      <p>
        Product the consumer identified in the complaint. Click on a product to expand sub-products
      </p>
      <div
        id="row-chart-product"
      />
    </div>
  </section>
  <section
    className="print-info-footer"
  >
    <p>
      <span>
        URL:
      </span>
       
      http://localhost/
    </p>
  </section>
</div>
`;

exports[`component:Results renders map panel without crashing 1`] = `
<div
  className="content_main map"
>
  <section
    className="map-panel"
  >
    <div>
      <summary
        className="action-bar"
        id="search-summary"
      >
        <div>
          <h2>
            Showing 
            1
             matches out of 
            100
             total complaints
          </h2>
        </div>
        <div>
          <h3
            className="h4 flex-all export-results"
          >
            <button
              className="a-btn a-btn__link export-btn"
              data-gtm_ignore="true"
              onClick={[Function]}
            >
              Export data
            </button>
            <button
              className="a-btn a-btn__link print-preview"
              onClick={[Function]}
            >
              <svg
                className="cf-icon-svg"
                viewBox="0 0 761.9 1200"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M203.7 889.9h355v50h-355zM203.7 788.4h355v50h-355z"
                />
                <path
                  d="M670.9 368.4L477.3 174.8H122.1c-16.5 0-30 13.5-30 30V500H70c-38.5 0-70 31.5-70 70v305.5c0 38.5 31.5 70 70 70h22.1v60.5c0 16.5 13.5 30 30 30h518.8c16.5 0 30-13.5 30-30v-60.5h21c38.5 0 70-31.5 70-70V569.9c0-38.5-31.5-70-70-70h-21V368.4zm-60 607.6H152.1V742.7h458.8V976zM425 234.8v146.4c0 21.1 17.3 38.4 38.4 38.4h147.5V500H152.1V234.8H425zm237.9 403.9c-16.6 0-30-13.4-30-30s13.4-30 30-30 30 13.4 30 30-13.4 30-30 30z"
                />
              </svg>
              Print
            </button>
          </h3>
        </div>
      </summary>
    </div>
    <div
      className="tabbed-navigation"
      data-tour="tabbed-navigation"
    >
      <section>
        <button
          className="trends tab"
          onClick={[Function]}
        >
          <svg
            className="cf-icon-svg"
            viewBox="0 0 15 19"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M13.55 15.256H1.45a.554.554 0 0 1-.553-.554V3.168a.554.554 0 1 1 1.108 0v10.98h11.544a.554.554 0 0 1 0 1.108zM3.121 13.02V6.888a.476.476 0 0 1 .475-.475h.786a.476.476 0 0 1 .475.475v6.132zm2.785 0V3.507a.476.476 0 0 1 .475-.475h.786a.476.476 0 0 1 .475.475v9.513zm2.785 0V6.888a.476.476 0 0 1 .475-.475h.786a.476.476 0 0 1 .475.475v6.132zm2.786 0v-2.753a.476.476 0 0 1 .475-.475h.785a.476.476 0 0 1 .475.475v2.753z"
            />
          </svg>
          Trends
        </button>
        <button
          className="list tab"
          onClick={[Function]}
        >
          <svg
            className="cf-icon-svg"
            viewBox="0 0 14 19"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M2.85 4.836a1.03 1.03 0 0 1-1.029 1.03h-.003a1.03 1.03 0 1 1 0-2.059h.003a1.03 1.03 0 0 1 1.03 1.03zm0 4.346a1.03 1.03 0 0 1-1.029 1.03h-.003a1.03 1.03 0 1 1 0-2.059h.003a1.03 1.03 0 0 1 1.03 1.03zm0 4.346a1.03 1.03 0 0 1-1.029 1.03h-.003a1.03 1.03 0 1 1 0-2.059h.003a1.03 1.03 0 0 1 1.03 1.03zm1.13-8.692a1.03 1.03 0 0 1 1.029-1.029h7.173a1.03 1.03 0 1 1 0 2.058H5.009a1.03 1.03 0 0 1-1.03-1.029zm9.231 4.347a1.03 1.03 0 0 1-1.029 1.029H5.009a1.03 1.03 0 1 1 0-2.059h7.173a1.03 1.03 0 0 1 1.03 1.03zm0 4.345a1.03 1.03 0 0 1-1.029 1.03H5.009a1.03 1.03 0 1 1 0-2.059h7.173a1.03 1.03 0 0 1 1.03 1.03z"
            />
          </svg>
          List
        </button>
        <button
          className="map tab active"
          onClick={[Function]}
        >
          <svg
            className="cf-icon-svg"
            viewBox="0 0 16 19"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15.084 15.2H.916a.264.264 0 0 1-.254-.42l2.36-4.492a.865.865 0 0 1 .696-.42h.827a9.51 9.51 0 0 0 .943 1.108H3.912l-1.637 3.116h11.45l-1.637-3.116h-1.34a9.481 9.481 0 0 0 .943-1.109h.591a.866.866 0 0 1 .696.421l2.36 4.492a.264.264 0 0 1-.254.42zM11.4 7.189c0 2.64-2.176 2.888-3.103 5.46a.182.182 0 0 1-.356 0c-.928-2.572-3.104-2.82-3.104-5.46a3.282 3.282 0 0 1 6.563 0zm-1.86-.005a1.425 1.425 0 1 0-1.425 1.425A1.425 1.425 0 0 0 9.54 7.184z"
            />
          </svg>
          Map
        </button>
      </section>
    </div>
    <div
      className="layout-row refine-bar"
    >
      <section
        className="filter-panel-toggle"
      >
        <div
          className="m-btn-group"
        >
          <p>
             
          </p>
          <button
            className="a-btn"
            onClick={[Function]}
          >
            Filter results
          </button>
        </div>
      </section>
      <span
        className="separator"
      />
      <section
        className="m-btn-group"
      >
        <p>
          Map shading
        </p>
        <button
          aria-label="Display map by complaints"
          className="a-btn "
          disabled={false}
          onClick={[Function]}
        >
          Complaints
        </button>
        <button
          aria-label="Display map by complaints per 1,000 people"
          className="a-btn a-btn__disabled"
          disabled={false}
          onClick={[Function]}
        >
          Complaints per 1,000 
          <span>
            population
          </span>
        </button>
      </section>
    </div>
    <div>
      <div
        className="cfpb-chart "
        data-chart-type="tile_map"
        id="tile-chart-map"
      />
    </div>
    <div
      className="map-toolbar"
    >
      <section
        className="state-heading"
      >
        <span>
          United States of America
        </span>
        <span>
          
        </span>
        
      </section>
      
    </div>
    <div
      className="row-chart-section"
    >
      <h3>
        Product by highest complaint volume 7/10/2017 to 7/10/2020
      </h3>
      <p>
        Product the consumer identified in the complaint. Click on a product to expand sub-products
      </p>
      <div
        id="row-chart-product"
      />
    </div>
  </section>
</div>
`;

exports[`component:Results renders trends panel without crashing 1`] = `
<div
  className="content_main trends"
>
  <section
    className="trends-panel"
  >
    <div>
      <summary
        className="action-bar"
        id="search-summary"
      >
        <div>
          <h2>
            Showing 
            1
             matches out of 
            100
             total complaints
          </h2>
        </div>
        <div>
          <h3
            className="h4 flex-all export-results"
          >
            <button
              className="a-btn a-btn__link export-btn"
              data-gtm_ignore="true"
              onClick={[Function]}
            >
              Export data
            </button>
            <button
              className="a-btn a-btn__link print-preview"
              onClick={[Function]}
            >
              <svg
                className="cf-icon-svg"
                viewBox="0 0 761.9 1200"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M203.7 889.9h355v50h-355zM203.7 788.4h355v50h-355z"
                />
                <path
                  d="M670.9 368.4L477.3 174.8H122.1c-16.5 0-30 13.5-30 30V500H70c-38.5 0-70 31.5-70 70v305.5c0 38.5 31.5 70 70 70h22.1v60.5c0 16.5 13.5 30 30 30h518.8c16.5 0 30-13.5 30-30v-60.5h21c38.5 0 70-31.5 70-70V569.9c0-38.5-31.5-70-70-70h-21V368.4zm-60 607.6H152.1V742.7h458.8V976zM425 234.8v146.4c0 21.1 17.3 38.4 38.4 38.4h147.5V500H152.1V234.8H425zm237.9 403.9c-16.6 0-30-13.4-30-30s13.4-30 30-30 30 13.4 30 30-13.4 30-30 30z"
                />
              </svg>
              Print
            </button>
          </h3>
        </div>
      </summary>
    </div>
    <div
      className="tabbed-navigation"
      data-tour="tabbed-navigation"
    >
      <section>
        <button
          className="trends tab active"
          onClick={[Function]}
        >
          <svg
            className="cf-icon-svg"
            viewBox="0 0 15 19"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M13.55 15.256H1.45a.554.554 0 0 1-.553-.554V3.168a.554.554 0 1 1 1.108 0v10.98h11.544a.554.554 0 0 1 0 1.108zM3.121 13.02V6.888a.476.476 0 0 1 .475-.475h.786a.476.476 0 0 1 .475.475v6.132zm2.785 0V3.507a.476.476 0 0 1 .475-.475h.786a.476.476 0 0 1 .475.475v9.513zm2.785 0V6.888a.476.476 0 0 1 .475-.475h.786a.476.476 0 0 1 .475.475v6.132zm2.786 0v-2.753a.476.476 0 0 1 .475-.475h.785a.476.476 0 0 1 .475.475v2.753z"
            />
          </svg>
          Trends
        </button>
        <button
          className="list tab"
          onClick={[Function]}
        >
          <svg
            className="cf-icon-svg"
            viewBox="0 0 14 19"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M2.85 4.836a1.03 1.03 0 0 1-1.029 1.03h-.003a1.03 1.03 0 1 1 0-2.059h.003a1.03 1.03 0 0 1 1.03 1.03zm0 4.346a1.03 1.03 0 0 1-1.029 1.03h-.003a1.03 1.03 0 1 1 0-2.059h.003a1.03 1.03 0 0 1 1.03 1.03zm0 4.346a1.03 1.03 0 0 1-1.029 1.03h-.003a1.03 1.03 0 1 1 0-2.059h.003a1.03 1.03 0 0 1 1.03 1.03zm1.13-8.692a1.03 1.03 0 0 1 1.029-1.029h7.173a1.03 1.03 0 1 1 0 2.058H5.009a1.03 1.03 0 0 1-1.03-1.029zm9.231 4.347a1.03 1.03 0 0 1-1.029 1.029H5.009a1.03 1.03 0 1 1 0-2.059h7.173a1.03 1.03 0 0 1 1.03 1.03zm0 4.345a1.03 1.03 0 0 1-1.029 1.03H5.009a1.03 1.03 0 1 1 0-2.059h7.173a1.03 1.03 0 0 1 1.03 1.03z"
            />
          </svg>
          List
        </button>
        <button
          className="map tab"
          onClick={[Function]}
        >
          <svg
            className="cf-icon-svg"
            viewBox="0 0 16 19"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15.084 15.2H.916a.264.264 0 0 1-.254-.42l2.36-4.492a.865.865 0 0 1 .696-.42h.827a9.51 9.51 0 0 0 .943 1.108H3.912l-1.637 3.116h11.45l-1.637-3.116h-1.34a9.481 9.481 0 0 0 .943-1.109h.591a.866.866 0 0 1 .696.421l2.36 4.492a.264.264 0 0 1-.254.42zM11.4 7.189c0 2.64-2.176 2.888-3.103 5.46a.182.182 0 0 1-.356 0c-.928-2.572-3.104-2.82-3.104-5.46a3.282 3.282 0 0 1 6.563 0zm-1.86-.005a1.425 1.425 0 1 0-1.425 1.425A1.425 1.425 0 0 0 9.54 7.184z"
            />
          </svg>
          Map
        </button>
      </section>
    </div>
    <div
      className="layout-row refine-bar"
    >
      <section
        className="filter-panel-toggle"
      >
        <div
          className="m-btn-group"
        >
          <p>
             
          </p>
          <button
            className="a-btn"
            onClick={[Function]}
          >
            Filter results
          </button>
        </div>
      </section>
      <section
        className="cf-select"
        data-tour="select-lens"
      >
        <label
          className="u-visually-hidden"
          htmlFor="select-lens"
        >
          Aggregate complaints by
        </label>
        <p>
          Aggregate by
        </p>
        <select
          id="select-lens"
          onChange={[Function]}
          value="Overview"
        >
          <option
            disabled={false}
            value="Overview"
          >
            Overview
          </option>
          <option
            disabled={false}
            value="Company"
          >
            Company
          </option>
          <option
            disabled={false}
            value="Product"
          >
            Product
          </option>
        </select>
      </section>
      <span
        className="separator"
      />
      <section
        className="cf-select"
        data-tour="select-interval"
      >
        <label
          className="u-visually-hidden"
          htmlFor="select-interval"
        >
          Choose the Date interval
        </label>
        <p>
          Date interval
        </p>
        <select
          id="select-interval"
          onChange={[Function]}
        >
          <option
            disabled={true}
            value="Day"
          >
            Day
          </option>
          <option
            disabled={false}
            value="Week"
          >
            Week
          </option>
          <option
            disabled={false}
            value="Month"
          >
            Month
          </option>
          <option
            disabled={false}
            value="Quarter"
          >
            Quarter
          </option>
          <option
            disabled={false}
            value="Year"
          >
            Year
          </option>
        </select>
      </section>
    </div>
  </section>
</div>
`;
